src/api.cpp

```diff
@@ -392,8 +411,18 @@ static int DataLoader_init(aeon_DataLoader* self, PyObject* args, PyObject* kwds
 static PyObject* aeon_reset(PyObject* self, PyObject*)
 {
     INFO << " aeon_reset";
-    DL_get_loader(self)->reset();
-    return Py_None;
+    try
+    {
+        DL_get_loader(self)->reset();
+        return Py_None;
+    }
+    catch (std::exception& e)
+    {
+        stringstream ss;
+        ss << "Error when resetting iterator: " << e.what();
+        PyErr_SetString(PyExc_RuntimeError, ss.str().c_str());
+        return NULL;
```

Comment:
Why not `Py_None` ?

Reply:
Because as I found and quoted earlier, error handling is a 2 stage process - setting PyErr_SetString and returning nullptr in case of pointer of -1 in case of integer.
>Most functions also return an error indicator, usually NULL if they are supposed to return a pointer, or -1 if they return an integer (exception: the PyArg_*() functions return 1 for success and 0 for failure). 

https://docs.python.org/2/c-api/exceptions.html